### ISSUE TITLE
Datadog logs shows source undefined

### ISSUE THREAD

**Issue by jenil** [vercel/integrations#77]
![image](https://user-images.githubusercontent.com/971411/79719475-a5733c80-82ac-11ea-9027-9514fa3da9a5.png)

**Comment by sioncojp** [comment]
I think it is correct to be tagged with `Source:vercel`.

**Comment by coetry** [comment]
Thank you for raising this issue! We can take care of that :+1:

**Comment by matthewdavidson** [comment]
Any update on this? 

Note in my case there is already a `source` attribute in the payload. Possible values I have seen are: `lambda` or `static`.

I believe the fix is to instead send this attribute as `ddsource` as [this is what datadog expects](https://docs.datadoghq.com/logs/processing/#source-attribute).

**Comment by hydrosquall** [comment]
@coetry wanted to check if this was already being looked into? I'm wasn't familiar with the usual Vercel project structure / how the LogDrain abstraction works so I might be off, but it looks like the `ddsource` key that @matthewdavidson identified could be added somewhere around here.

https://github.com/vercel/integrations/blob/b94b2a8b12bfe6a0a6a8f36d577d1e0d2e0f1ed6/datadog-logs/lib/setup.js#L24-L38

**Comment by nkzawa** [comment]
To add the `source` tag to datadog, it requires to add `ddsource` to the request body according to their documents https://docs.datadoghq.com/api/latest/logs, but log drain does not support customizing data format, therefore we cannot set the tag for now.

Sorry for the inconvenience but we are trying to fix this problem.

**Comment by lukeshumard** [comment]
Hi @nkzawa, I'm also working on solving this. I'm pretty certain that by adding `ddsource=vercel` as a query string parameter to the `url` value when creating the Log Drain would resolve the issue.

`https://http-intake.logs.${host}/v1/input/${encodeURIComponent(key)}?ddsource=vercel`

I don't believe there's any need to customize the data formatting, but I could be missing something here.

**Comment by lukeshumard** [comment]
I've tried out the suggestion above and it works. Datadog picks up the `ddsource` and `service` as query string parameters when specifying the URL for the Log Drain.

```
https://http-intake.logs.${datadog_region}/v1/input/${datadog_api_key}?ddsource=vercel
```
<img width="269" alt="Screen Shot 2021-02-25 at 10 52 44 PM" src="https://user-images.githubusercontent.com/472555/109230126-43cd0a00-77bc-11eb-9f59-ae271068b685.png">

I've also set the `service` by adding `?ddsource=vercel&service=datadog-client-test`. This means that the `VERCEL_URL` or deployment url gets set in Datadog as `host` as expected.

<img width="859" alt="Screen Shot 2021-02-25 at 10 57 46 PM" src="https://user-images.githubusercontent.com/472555/109230560-f69d6800-77bc-11eb-853b-94b282659364.png">

**Comment by nkzawa** [comment]
@lukeshumard I also tested that and it seems works fine, thank you for the suggestion 👍  I have been thinking it works only if log's `content-type` is `text/plain`. I will fix the integration to include the query parameter.

**Comment by matthewdavidson** [comment]
I'd like to draw attention to my previous comment https://github.com/vercel/integrations/issues/77#issuecomment-768973183

The log drain integration today already sends a `source` attribute with one of the following three values: `lambda`, `static` & `build`. I'd argue that there values are more correct than sending `vercel` as the `ddsource`, because the _source_  should corresponding to the integration technology, from datadogs own [docs](https://docs.datadoghq.com/logs/log_collection/?tab=host#attributes-and-tags):

<img width="400" alt="Screenshot 2021-02-26 at 08 36 20" src="https://user-images.githubusercontent.com/1440090/109276294-bf0adc00-780d-11eb-8c81-6f7b4b74a56c.png">

In fact - I am currently relying on the `source` attribute to filter between the three log types above (`lambda`, `static` & `build`). Rationale being that I am not interested in seeing build logs or static content requests, but I am interested in logs that come from serverless functions.

I use this filtering in conjunction with datadog's log preprocessing functionality which allows me to assign pre-determined log pipelines to automatically parse the logs and enrich them (e.g. assign the correct log severity etc). 

Furthermore - there exists a prebuilt pipeline processor for logs that come with a source of `lambda`:

![image](https://user-images.githubusercontent.com/1440090/109277067-b535a880-780e-11eb-8457-d99ec071c3e8.png)

In short, I believe https://github.com/vercel/integrations/pull/151 to be the wrong appraoch - instead we should be sending either `lambda`, `static` & `build` as the `ddsource` attribute.

**Comment by nkzawa** [comment]
@matthewdavidson Thank you for your suggestion and sorry that I didn't notice your comment. We're considering what is the best solution. #151 is not released yet.

**Comment by lukeshumard** [comment]
Hey @matthewdavidson, sorry I missed your comment earlier. Ideally this should be configurable to some degree. In my use case, I want Datadog to have both logs from Heroku and Vercel. In the Heroku setup, Datadog says to specify `ddsource=heroku`. In the documentation you've highlighted, it states...
> ...When [source] matches an integration name, Datadog automatically installs the corresponding parsers and facets.

Because of this, I don't think having `vercel` as the source is necessarily the wrong approach. However, it should be possible in Datadog for you to configure to filter between `lambda`, `static`, or `build` logs. I was able to setup use Remapper in my Log Pipeline to create a new attribute called `vercel_source`. I then used this attribute to create a Facet which I can now query.

<img width="1154" alt="Screen Shot 2021-02-26 at 12 09 08 PM" src="https://user-images.githubusercontent.com/472555/109298485-942f8080-782b-11eb-962f-58e220c87931.png">

<img width="961" alt="Screen Shot 2021-02-26 at 12 08 10 PM" src="https://user-images.githubusercontent.com/472555/109298506-9abdf800-782b-11eb-8a18-dea978958996.png">

**Comment by matthewdavidson** [comment]
Hey @lukeshumard. You make a good point. I think perhaps I am misunderstanding how datadog maps a `ddsource` string to a corresponding integration pipeline. I checked and there is indeed an official heroku integration pipeline mapped to a source value of `heroku` 👍 

Perhaps @nkzawa you could reach out to someone from datadog to figure out how we get an official vercel integration pipeline in datadog out of the box? That would save folks like @lukeshumard and myself from having to define custom pipelines ourselves.

**Comment by nkzawa** [comment]
I agree @lukeshumard and it'd be good to set `ddsource: vercel`. I will release the change.
We're planning for improving integration with datadog so stay tuned 🙏
